src/dmd/astbase.d

```diff
@@ -3995,13 +4007,165 @@ struct ASTBase
             t.isproperty = isproperty;
             t.isref = isref;
             t.isreturn = isreturn;
-            t.isscope = isscope;
-            t.iswild = iswild;
+            t.isthisscope = isthisscope;
+            t.isreturninferred = isreturninferred;
+            t.isscopeinferred = isscopeinferred;
+            t.isInOutParam = isInOutParam;
+            t.isInOutQual = isInOutQual;
             t.trust = trust;
             t.fargs = fargs;
             return t;
         }
 
+        /// set or get if the function has the `nothrow` attribute
+        bool isnothrow() const pure nothrow @safe @nogc
+        {
+            return (funcFlags & FunctionFlag.isnothrow) != 0;
+        }
+        /// ditto
+        void isnothrow(bool v) pure nothrow @safe @nogc
+        {
+            if (v) funcFlags |= FunctionFlag.isnothrow;
+            else funcFlags &= ~FunctionFlag.isnothrow;
+        }
+
+        /// set or get if the function has the `@nogc` attribute
+        bool isnogc() const pure nothrow @safe @nogc
+        {
+            return (funcFlags & FunctionFlag.isnogc) != 0;
+        }
+        /// ditto
+        void isnogc(bool v) pure nothrow @safe @nogc
+        {
+            if (v) funcFlags |= FunctionFlag.isnogc;
+            else funcFlags &= ~FunctionFlag.isnogc;
+        }
+
+        /// set or get if the function has the `@property` attribute
+        bool isproperty() const pure nothrow @safe @nogc
+        {
+            return (funcFlags & FunctionFlag.isproperty) != 0;
+        }
+        /// ditto
+        void isproperty(bool v) pure nothrow @safe @nogc
+        {
+            if (v) funcFlags |= FunctionFlag.isproperty;
+            else funcFlags &= ~FunctionFlag.isproperty;
+        }
+
+        /// set or get if the function has the `ref` attribute
+        bool isref() const pure nothrow @safe @nogc
+        {
+            return (funcFlags & FunctionFlag.isref) != 0;
+        }
+        /// ditto
+        void isref(bool v) pure nothrow @safe @nogc
+        {
+            if (v) funcFlags |= FunctionFlag.isref;
+            else funcFlags &= ~FunctionFlag.isref;
+        }
+
+        /// set or get if the function has the `return` attribute
+        bool isreturn() const pure nothrow @safe @nogc
+        {
+            return (funcFlags & FunctionFlag.isreturn) != 0;
+        }
+        /// ditto
+        void isreturn(bool v) pure nothrow @safe @nogc
+        {
+            if (v) funcFlags |= FunctionFlag.isreturn;
+            else funcFlags &= ~FunctionFlag.isreturn;
+        }
+
+        /// set or get if the function has the `scope` attribute
+        bool isthisscope() const pure nothrow @safe @nogc
```

Comment:
Might as well use camelCase since this introduces a new symbol which breaks the naming convention.

Reply:
I'll go for `isScopeQual`, as there's l`IsInOutQuall` already.